src/relax/transform/default_schedule.cc

```diff
@@ -0,0 +1,145 @@
+          // only fuse data parallel loops
+          for (size_t i = 0; i < loops.size(); ++i) {
+            if (iters[i]->iter_type == tir::IterVarType::kDataPar) {
+              data_parallel_loops.push_back(loops[i]);
+            }
+          }
+          if (data_parallel_loops.size() == 0) {
+            continue;
+          }
+          // fuse all data parallel loops
+          tir::LoopRV fused = sch->Fuse(data_parallel_loops, /*preserve_unit_iters=*/false);
+          int64_t product = std::numeric_limits<int64_t>::max();
+          if (sch->Get(fused)->extent->IsInstance<tir::IntImmNode>()) {
+            product = sch->Get(fused)->extent.as<tir::IntImmNode>()->value;
+          }
+          static const int64_t max_threadblocks = 256;
+          // schedule the fused loop
+          if (product > max_thread_per_block * max_threadblocks) {
+            Array<tir::LoopRV> splits = sch->Split(
+                fused,
+                /*factors=*/{NullOpt, Integer(max_threadblocks), Integer(max_thread_per_block)});
+            sch->Reorder(/*ordered_loop_rvs=*/{splits[1], splits[2], splits[0]});
+            sch->Bind(splits[1], "blockIdx.x");
+            sch->Bind(splits[2], "threadIdx.x");
+          } else {
+            Array<tir::LoopRV> splits = sch->Split(
+                fused, /*factors=*/{NullOpt, Integer(std::min(product, max_thread_per_block))});
+            sch->Bind(splits[0], "blockIdx.x");
+            sch->Bind(splits[1], "threadIdx.x");
+          }
+        }
+        mutator.builder_->AddFunction(sch->mod()->Lookup(gv->name_hint), gv->name_hint);
+      } else {
+        mutator.builder_->AddFunction(func, gv->name_hint);
+      }
+    }
+    return mutator.builder_->GetContextIRModule();
+  }
+
+ private:
+  explicit ThreadBindMutator(const IRModule& mod) : mod_(mod) {}
+
+ private:
+  /*! \brief The IRModule */
+  const IRModule& mod_;
+};
+
+IRModule DefaultSchedule(IRModule mod, int64_t max_thread_per_block) {
+  mod = ThreadBindMutator::Transform(mod, max_thread_per_block);
+  return mod;
+}
+
+namespace transform {
+
+Pass DefaultSchedule() {
+  runtime::TypedPackedFunc<IRModule(IRModule, PassContext)> pass_func =  //
+      [=](IRModule m, PassContext pc) {
+        tvm::Target target = tvm::Target::Current();
+        Integer max_thread_per_block = target->GetAttr<Integer>("max_num_threads").value_or(-1);
+        if (target->kind->name != "cuda") {
+          ICHECK_NE(max_thread_per_block, -1) << "max_num_threads is not set for target " << target;
+          return m;
+        }
+        return relax::DefaultSchedule(m, max_thread_per_block.IntValue());
+      };
+  return CreateModulePass(/*pass_function=*/pass_func,      //
+                          /*opt_level=*/0,                  //
+                          /*pass_name=*/"DefaultSchedule",  //
+                          /*required=*/{});
+}
+
+TVM_REGISTER_GLOBAL("relax.transform.DefaultSchedule").set_body_typed(DefaultSchedule);
```

Comment:
I feel it would be better to move it into `tir.transform.DefaultSchedule` (src/tir/transform), looks only PrimFunc is updated